src/libs/auth/oidcBroker.ts

```diff
@@ -0,0 +1,93 @@
+import { IdTokenClaims, OidcMetadata, SigninPopupArgs, SigninSilentArgs, User, UserManager, UserManagerSettings, WebStorageStateStore } from 'oidc-client-ts';
+import { Config } from '../config';
+import { OAuth2, OAuthConfig } from '../ajax/OAuth2';
+
+// Our config for b2C claims are defined here: https://github.com/broadinstitute/terraform-ap-deployments/tree/master/azure/b2c/policies
+// The standard b2C claims are defined here: https://learn.microsoft.com/en-us/azure/active-directory/develop/id-token-claims-reference
+export interface B2cIdTokenClaims extends IdTokenClaims {
+  email_verified?: boolean;
+  idp?: string;
+  idp_access_token?: string;
+  tid?: string;
+  ver?: string;
+}
+
+export interface OidcUser extends User {
+  profile: B2cIdTokenClaims;
+}
+
+type OidcUserManager = UserManager;
+
+let config: OAuthConfig | null = null;
+let userManagerSettings: UserManagerSettings | null = null;
+let userManager: UserManager | null = null;
+
+const generateOidcUserManagerSettings = async (
+  config: OAuthConfig
+): Promise<UserManagerSettings> => {
+  const metadata: Partial<OidcMetadata> = {
+    authorization_endpoint: `${await Config.getApiUrl()}/oauth2/authorize`,
+    token_endpoint: `${await Config.getApiUrl()}/oauth2/token`,
+  };
+  return {
+    authority: config.authorityEndpoint,
+    client_id: config.clientId,
+    popup_redirect_uri: `${window.origin}/redirect-from-oauth`,
+    silent_redirect_uri: `${window.origin}/redirect-from-oauth-silent`,
+    metadata,
+    prompt: 'consent login',
+    scope: 'openid email profile',
```

Comment:
Re my above comment: I see `prompt=login` is already set here. We'd need to figure out how to set the clientId as a scope if not using the custom oauth2 endpoints. (Or validate that this is still a requirement in B2C?)